geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/Destroy65.java

```diff
@@ -111,7 +112,30 @@ public void cmdExecute(final Message clientMessage, final ServerConnection serve
 
     final Operation operation;
     try {
-      operation = getOperation(clientMessage.getPart(3), Operation.DESTROY);
+      final Part operationPart = clientMessage.getPart(3);
+
+      if (operationPart.isBytes()) {
+        final byte[] bytes = operationPart.getSerializedForm();
+        if (null == bytes || 0 == bytes.length) {
+          // older clients can send empty bytes for default operation.
+          operation = Operation.DESTROY;
+        } else {
+          operation = Operation.fromOrdinal(bytes[0]);
+        }
+      } else {
+        // Fallback for older clients.
+        if (operationPart.getObject() == null) {
```

Comment:
Maybe cache the result of operationPart.getObject().  That operation will deserialize the bytes every time you invoke it.